packages/autoloader/src/functions.php

```diff
@@ -42,16 +51,43 @@ function autoloader( $class_name ) {
  * @param Version_Selector $version_selector The Version_Selector object.
  */
 function enqueue_files( $plugins_handler, $version_selector ) {
-	require_once __DIR__ . '/class-classes-handler.php';
-	require_once __DIR__ . '/class-files-handler.php';
+	require_once __DIR__ . '/class-manifest-handler.php';
+	require_once __DIR__ . '/class-version-loader.php';
 
-	$classes_handler = new Classes_Handler( $plugins_handler, $version_selector );
-	$classes_handler->set_class_paths();
+	$manifest_handler = new Manifest_Handler( $plugins_handler, $version_selector );
 
-	$files_handler = new Files_Handler( $plugins_handler, $version_selector );
-	$files_handler->set_file_paths();
+	global $jetpack_packages_psr4;
+	$manifest_handler->register_plugin_manifests(
+		'vendor/composer/jetpack_autoload_psr4.php',
+		$jetpack_packages_psr4
+	);
```

Comment:
If we don't generate the PSR-4 file in optimized builds we could use the autoloader version comparison to also detect if an autoloader is in an unoptimized state to skip this manifest check, what do you think?

Reply:
Yeah, I think that makes sense.

Reply:
Since the `Manifest_Handler` does an `is_readable()` check before actually loading the manifest we probably can't do much more than we already are. One option might be to add a constant that can be used for optimization to entirely disable PSR-4 autoloading?

I've also adjusted the hander to only look up the active plugin paths once.